R/app-server.R

```diff
@@ -187,13 +195,46 @@ app_server <- function(input, output, session) {
         fail_msg = "Some values in the assay metadata are invalid"
       )
     })
+
+    # Individual and specimen IDs are not duplicated ---------------------------
     duplicate_indiv_ids <- reactive({
       check_indiv_ids_dup(indiv())
     })
     duplicate_specimen_ids <- reactive({
       check_specimen_ids_dup(biosp())
     })
 
+    # Empty columns produce warnings -------------------------------------------
+    empty_cols_manifest <- reactive({
+      check_cols_empty(
+        manifest(),
+        success_msg = "No columns are empty in the manifest",
+        fail_msg = "Some columns are empty in the manifest"
```

Comment:
How is a warning message coded?

Reply:
Looks like if the argument `strict = FALSE` is in `check_cols_empty`, a warning is thrown instead of an error.

Reply:
Correct: by default `check_cols_empty()` creates a warning object for empty columns, but if `strict = TRUE` then it creates an error object. The `fail_msg` is the same in either case.